src/source/PeerConnection/PeerConnection.c

```diff
@@ -84,12 +85,20 @@ VOID onInboundPacket(UINT64 customData, PBYTE buff, UINT32 buffLen)
             CHK_STATUS(putSctpPacket(pKvsPeerConnection->pSctpSession, buff, signedBuffLen));
         }
 
-        if (pKvsPeerConnection->pSrtpSession == NULL) {
-            dtlsSessionIsInitFinished(pKvsPeerConnection->pDtlsSession, &isDtlsConnected);
-            if (isDtlsConnected) {
-                allocateSrtp(pKvsPeerConnection);
+        CHK_STATUS(dtlsSessionIsInitFinished(pKvsPeerConnection->pDtlsSession, &isDtlsConnected));
+        if (isDtlsConnected) {
+            MUTEX_LOCK(pKvsPeerConnection->peerConnectionObjLock);
+            locked = TRUE;
+            if (pKvsPeerConnection->pSrtpSession == NULL) {
+                CHK_STATUS(allocateSrtp(pKvsPeerConnection));
+            }
+
+            if (pKvsPeerConnection->pSctpSession == NULL) {
                 CHK_STATUS(allocateSctp(pKvsPeerConnection));
             }
+
+            MUTEX_UNLOCK(pKvsPeerConnection->peerConnectionObjLock);
```

Comment:
Do we need to drop the lock? Looking at the code nothing else happens.

Do you anticipate lock contention?

Reply:
I'm using the lock to avoid duplicate allocateSrtp and allocateSctp calls, so I dont need to hold it anymore after allocate calls are done.